src/ui/CodingContractPopup.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react";
+import { KEY } from "../../utils/helpers/keyCodes";
+
+import { CodingContract, CodingContractType, CodingContractTypes } from "../CodingContracts";
+import { ClickableTag, CopyableText } from "../ui/React/CopyableText";
+import { PopupCloseButton } from "../ui/React/PopupCloseButton";
+
+type IProps = {
+    c: CodingContract;
+    onCloseClick: () => void;
+    onSolveClick: () => void;
+}
+
+export class CodingContractPopup extends React.Component<IProps>{
+    state: any;
+    constructor(props: IProps) {
+        super(props);
+        this.state = { value: ''};
+        this.setValue = this.setValue.bind(this);
+    }
+    getValue(){
+        let input = document.getElementById('contractInput')?.getAttribute('value');
```

Comment:
See https://github.com/danielyxie/bitburner/blob/master/src/Faction/ui/DonateOption.tsx for example on how to bind input value to state.